test/e2e/specs/editor/various/splitting-merging.spec.js

```diff
@@ -255,6 +255,26 @@ test.describe( 'splitting and merging blocks (@firefox, @webkit)', () => {
 		expect( content ).toBe( `` );
 	} );
 
+	test( 'should not merge heading with empty paragraph block on backspace', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await editor.insertBlock( { name: 'core/paragraph' } );
+		await editor.insertBlock( { name: 'core/heading' } );
+		await page.keyboard.type( 'Heading' );
+		await pageUtils.pressKeys( 'ArrowLeft', { times: 7 } );
```

Comment:
For tests, it's generally enough to just type one character (e.g. "a")

Reply:
Sorted!